LogHandler.groovy

```diff
@@ -0,0 +1,72 @@
+// -----------------------------------------------
+// This class holds all of the current search logs
+// -----------------------------------------------
+class LogHandler {
+
+    LogHandler(List<SearchLog> pOldLogs = [], List<SearchLog> pSearchLogs = [], List<RequestLog> pRequestLogs = []){
+        this.oldLogs         = pOldLogs
+        this.searchLogs      = pSearchLogs
+        this.requestLogs     = pRequestLogs
+        this.oldLogCount     = pOldLogs.size()
+        this.searchLogCount  = pSearchLogs.size()
+        this.requestLogCount = pRequestLogs.size()
+    }
+
+    // Getters/Setters
+    // ---------------
+
+    void get_logs_from_parser(LogParser parser){
+        this.oldLogs = parser.pass_old_search_logs()
```

Comment:
This is actually confusing, because it doesn't just actually pass the old search logs. The method also reinitializes the search logs to an empty array. It would probably better to say `this.oldLogs = parser.searchLogs`, then have a method that calls the parser to reinitializes those fields.